pkg/test/env/istio.go

```diff
@@ -104,10 +101,15 @@ var (
 	ServiceAccountFilePath = path.Join(ChartsDir, "helm-service-account.yaml")
 
 	// RedisInstallFilePath is the redis installation file.
-	RedisInstallFilePath = path.Join(IstioRoot, "pkg/test/framework/components/redis/redis.yaml")
+	RedisInstallFilePath = path.Join(IstioSrc, "pkg/test/framework/components/redis/redis.yaml")
 )
 
 func getDefaultIstioTop() string {
+	// If running in build container, use expected build container path "/work"
```

Comment:
This is a code smell imo. Build container should be like Istio -- your application doesn't need to know its there or detect it and behave differently, you just get free value on top.

I suggest:

```diff
diff --git a/pkg/test/env/istio.go b/pkg/test/env/istio.go
index 2c3599aaf..3a5cda003 100644
--- a/pkg/test/env/istio.go
+++ b/pkg/test/env/istio.go
@@ -105,6 +105,10 @@ var (
 )

 func getDefaultIstioTop() string {
+       root, f := os.LookupEnv("REPO_ROOT")
+       if f {
+               return root
+       }
        // If running in build container, use expected build container path "/work"
        inBuildContainer := os.Getenv("IN_BUILD_CONTAINER")
        if inBuildContainer == "1" {
```

and remove what you added. I think this covers all cases:
* container on/off through makefile
* running through IDE (eg no container)

Reply:
Oh my bad, I see the issue. this is REPO_ROOT/../../.. basically?

Looks like we never actually use top except to determine `IstioSrc` though. So I would suggest you do what I mentioned, except make this `getIstioSrc` or something, if that makes sense?

Let me know if you want me to send commits, I know you are on a boat soon/now

Reply:
thanks for the suggestion - makes sense.